src/hooks.ts

```diff
@@ -1,13 +1,35 @@
+import React from 'react';
 import AsyncStorage from './AsyncStorage';
-import type { AsyncStorageHook } from './types';
 
-export function useAsyncStorage(key: string): AsyncStorageHook {
-  return {
-    getItem: (...args) => AsyncStorage.getItem(key, ...args),
-    setItem: (...args) => AsyncStorage.setItem(key, ...args),
-    mergeItem: (...args) =>
+export function useAsyncStorage(key: string) {
+  const getItem = React.useCallback(
+    (...args) => AsyncStorage.getItem(key, ...args),
+    [key]
+  );
+
+  const setItem = React.useCallback(
+    //@ts-ignore
```

Comment:
We shouldn't add any `@ts-ignore`. What's the error here? Can we fix it?

Reply:
Yes its required because TS complains:
[ts] src/hooks.ts(11,44): error TS2556: A spread argument must either have a tuple type or be passed to a rest parameter.